### PR TITLE
implement evaluate_to rule in eval

### PR DESCRIPTION
**Pull request by adrianlekko**

### CODE REVIEW

File: internal/memory/store.go [hunk 1]
```diff
@@ -158,12 +160,36 @@ func (s *store) getCommitSha() string {
...
 	cfg, err := s.get(namespace, key)
 	if err != nil {
 		return nil, nil, err
 	}
-	evaluableConfig := eval.NewV1Beta3(cfg.Config, namespace)
+
+	// check metadata in cfg to see if there is a reference config needs to evaluate
+	// https://github.com/lekkodev/config-test/pull/489/files
+	// metadata = {
+	//     "segments": {
```
Comment (adrianlekko) [inline internal/memory/store.go:173]: metadata = {
	   ["flag1", "flag2", ...]
	// },

File: internal/memory/store.go [hunk 2]
```diff
@@ -158,12 +160,38 @@ func (s *store) getCommitSha() string {
...
+
+	referencedConfigToValueMap := make(map[string]interface{})
+	referencedConfigKeys := s.getReferencedConfigKeys(cfg)
+	if len(referencedConfigKeys) > 0 {
+		for configKey := range referencedConfigKeys {
+			referencedDest := &wrapperspb.StringValue{}
+			_, _, err2 := s.evaluateType(configKey, namespace, lc, referencedDest)
+			if err2 != nil {
+				return nil, nil, err2
+			}
```
Comment (passichenko) [inline internal/memory/store.go:189]:
```suggestion
			_, _, err := s.evaluateType(configKey, namespace, lc, referencedDest)
			if err != nil {
				return nil, nil, err
			}
```

Comment (passichenko) [reply internal/memory/store.go:189]: in go it's fine to always redefine `err`

File: pkg/rules/functions.go [hunk 3]
```diff
@@ -23,6 +23,18 @@ import (
 	"github.com/pkg/errors"
 )
 
+func (v1b3 *v1beta3) evaluateEvaluateTo(
+	bucketF *rulesv1beta3.CallExpression_EvaluateTo,
```
Comment (passichenko) [inline pkg/rules/functions.go:27]:
```suggestion
	evaluateToF *rulesv1beta3.CallExpression_EvaluateTo,
```

**Pull request merged**